packages/fx-core/src/component/driver/teamsApp/collaboration.ts

```diff
@@ -0,0 +1,138 @@
+    teamsAppId: string,
+    userInfo: AppUser
+  ): Promise<Result<ResourcePermission[], FxError>> {
+    try {
+      const appStudioTokenRes = await this.tokenProvider.getAccessToken({
+        scopes: AppStudioScopes,
+      });
+      const appStudioToken = appStudioTokenRes.isOk() ? appStudioTokenRes.value : undefined;
+
+      await AppStudioClient.grantPermission(teamsAppId, appStudioToken as string, userInfo);
+      const result: ResourcePermission[] = [
+        {
+          name: Constants.PERMISSIONS.name,
+          roles: [Constants.PERMISSIONS.admin],
+          type: Constants.PERMISSIONS.type,
+          resourceId: teamsAppId,
+        },
+      ];
+      return ok(result);
+    } catch (error) {
+      return err(this.handleError(error, ctx));
+    }
+  }
+
+  @hooks([addStartAndEndTelemetry(EventName.listCollaborator, componentName)])
+  public async listCollaborator(
+    ctx: Context,
+    teamsAppId: string
+  ): Promise<Result<TeamsAppAdmin[], FxError>> {
+    try {
+      const appStudioTokenRes = await this.tokenProvider.getAccessToken({
+        scopes: AppStudioScopes,
+      });
+      const appStudioToken = appStudioTokenRes.isOk() ? appStudioTokenRes.value : undefined;
+
+      const userLists = await AppStudioClient.getUserList(teamsAppId, appStudioToken as string);
+      if (!userLists) {
+        return ok([]);
+      }
+
+      const teamsAppAdmin: TeamsAppAdmin[] = userLists
+        .filter((userList) => {
+          return userList.isAdministrator;
+        })
+        .map((userList) => {
+          return {
+            userObjectId: userList.aadId,
+            displayName: userList.displayName,
+            userPrincipalName: userList.userPrincipalName,
+            resourceId: teamsAppId,
+          };
+        });
+
+      return ok(teamsAppAdmin);
+    } catch (error) {
+      return err(this.handleError(error, ctx));
+    }
+  }
+
+  @hooks([addStartAndEndTelemetry(EventName.checkPermission, componentName)])
+  public async checkPermission(
+    ctx: Context,
+    teamsAppId: string,
+    userInfo: AppUser
+  ): Promise<Result<ResourcePermission[], FxError>> {
+    try {
+      const appStudioTokenRes = await this.tokenProvider.getAccessToken({
+        scopes: AppStudioScopes,
+      });
+      const appStudioToken = appStudioTokenRes.isOk() ? appStudioTokenRes.value : undefined;
+
+      const teamsAppRoles = await AppStudioClient.checkPermission(
+        teamsAppId,
+        appStudioToken as string,
+        userInfo.aadId
+      );
+
+      const result: ResourcePermission[] = [
+        {
+          name: Constants.PERMISSIONS.name,
+          roles: [teamsAppRoles as string],
+          type: Constants.PERMISSIONS.type,
+          resourceId: teamsAppId,
+        },
+      ];
+      return ok(result);
+    } catch (error) {
+      return err(this.handleError(error, ctx));
+    }
+  }
+
+  private handleError(error: any, ctx: Context): FxError {
+    if (axios.isAxiosError(error)) {
```

Comment:
Does AppStudioClient have same implementation pattern as AadAppClient? I'm just wondering whether same error hanlding logic can apply to AppStudioClient.